File: STJS/HelloWorld.java
public class HelloWorld {
	public static void main(String[] args) {
		window.onload = new Callback1<DOMEvent>() {
			@Override
			public void $invoke(DOMEvent ev) {
				Form form = window.document.forms.$get(0);
				Element button = form.elements.$get("say");
				final Input text = form.elements.$get("to");
				button.onclick = new Function1<DOMEvent, Boolean>() {
					@Override
					public Boolean $invoke(DOMEvent ev) {
						alert("Hello " + text.value);
						return true;
					}
				};
			}
		};
	}
}

File: STJS/helloworld.js
var HelloWorld = function(){};
stjs.extend(HelloWorld, null, [], function(constructor, prototype){
    constructor.main = function(args) {
        window.onload = function(ev) {
            var form = window.document.forms[0];
            var button = form.elements["say"];
            var text = form.elements["to"];
            button.onclick = function(ev) {
                alert("Hello " + text.value);
                return true;
            };
        };
    };
}, {});

if (!stjs.mainCallDisabled) HelloWorld.main();
